lib/identity_cache/query_api.rb

```diff
@@ -476,5 +484,15 @@ def was_new_record? # :nodoc:
       pk = self.class.primary_key
       !destroyed? && transaction_changed_attributes.has_key?(pk) && transaction_changed_attributes[pk].nil?
     end
+
+    def load_and_readonlyify(association_name)
+      record_or_records = send(association_name)
+      if self.class.reflect_on_association(association_name).collection?
+        record_or_records.map(&:readonly!)
+      else
+        record_or_records.try(:readonly!)
```

Comment:
This method isn't ever for records loaded from the cache.  I think the only purpose is to try to make code that uses `fetch_#{association}` fail deterministically if the records could have been loaded from the cache.  However, in some cases it is marking records as read-only that could have been deterministically loaded from the database, my example tries to show.